### ISSUE TITLE
[default-param-last] False positive for optional arguments denoted by `?`

### ISSUE THREAD

**Issue by buschtoens** [typescript-eslint/typescript-eslint#1407]
**Repro**

```JSON
{
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint"],
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended"
  ]
}
```

```TS
function foo(a: string, b = 'b', c?: string) {}
```

**Expected Result**

The code should pass linting, as `c?` is equivalent to `c = undefined`.

**Actual Result**

`b` fails `default-param-last`, as it is not the last parameter.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.14.0` |
| `@typescript-eslint/parser`        | `2.14.0` |
| `TypeScript`                       | `3.7.3` |
| `ESLint`                           | `6.7.2` |
